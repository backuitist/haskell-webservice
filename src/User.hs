module User where

import Protolude

import Data.Aeson (ToJSON, FromJSON)
import Data.Text (Text)
import GHC.Generics (Generic)

data User = User
  { name :: Text
  , age :: Int
  } deriving (Eq, Show, Generic)

-- Use Aeson to convert user to JSON
instance ToJSON User
instance FromJSON User