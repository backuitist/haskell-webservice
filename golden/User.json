{
    "seed": 4151032124030248282,
    "samples": [
        {
            "age": 10,
            "name": "\\ò£ªªw\u0010vyôƒ›–z~ò®‡½aE"
        },
        {
            "age": -25,
            "name": "ó³©–a^Cðµ¤µj\u0011ñ²¥…:5`6"
        },
        {
            "age": -2,
            "name": "1E\u001fv6p"
        },
        {
            "age": -18,
            "name": "ò©¨µó¾’š?\u00171tHpð·†Œ\u0006&ð•µ­"
        },
        {
            "age": -5,
            "name": "ó²·–o.5\u0017F"
        }
    ]
}